pkg/agent/conversation.go

```diff
@@ -222,108 +223,114 @@ func (a *Conversation) RunOneRound(ctx context.Context, query string) error {
 			agentTextBlock.SetStreaming(false)
 		}
 
-		// TODO(droot): Run all function calls in parallel
-		// (may have to specify in the prompt to make these function calls independent)
-		for _, call := range functionCalls {
-			toolCall, err := a.Tools.ParseToolInvocation(ctx, call.Name, call.Arguments)
-			if err != nil {
-				return fmt.Errorf("building tool call: %w", err)
-			}
-
-			// Check if this is a kubectl or bash command that might be interactive
-			isInteractive := false
-			var errMsg string
-			if call.Name == "kubectl" || call.Name == "bash" {
-				if command, ok := call.Arguments["command"].(string); ok {
-					isInteractive, errMsg = tools.IsInteractiveCommand(command)
+		// Only process function calls if not using tool-use shim
+		if !a.EnableToolUseShim {
```

Comment:
We don't need to condition. We have functionCalls for both (tool-use or non-tool-use API) at this point.